### PR TITLE
Add links to project files and folders in README.md

### PR DESCRIPTION
**Pull request by hdurix**
Add links to relative paths in project README file. 

For example on Github, or Gitlab, it creates real links on your project, see [there](https://github.com/hdurix/jhipster-readme-with-links).

### CODE REVIEW

File: generators/server/templates/_README.md [hunk 1]
```diff
@@ -126,7 +126,7 @@ To optimize the <%= baseName %> application for production, run:
     ./gradlew -Pprod clean bootRepackage<% } %>
 
 <%_ if(!skipClient) { _%>
-This will concatenate and minify the client CSS and JavaScript files. It will also modify `index.html` so it references these new files.
+This will concatenate and minify the client CSS and JavaScript files. It will also modify [index.html](index.html) so it references these new files.
```
Comment (pascalgrimaud) [inline generators/server/templates/_README.md]: This link doesn't exist. It should be `src/main/webapp/index.html` instead
Can you do another PR with [ci skip] in your commit message? So it won't trigger a new Travis build :construction_worker_man: :construction_worker_woman:

Comment (gmarziou) [inline generators/server/templates/_README.md]: Sorry I did not catch it in my review, thanks Pascal. I'm wondering  whether we should even have a link here as the file may be built by webpack so only exists in target/www folder.

Comment (pascalgrimaud) [inline generators/server/templates/_README.md]: np :)
I agree, it's not mandatory to have it in README

Comment (hdurix) [inline generators/server/templates/_README.md]: Ok I remove the link for `index.html` in a new PR.

**Pull request merged**